ISO C++17 does not allow 'register' keyword

Compiling STIR from source on Mac OSX 10.15.7 using the C++17 standard, I ran into this error:

```bash
/Users/jvavrek/STIR/src/display/display_array.cxx:233:3: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
  register  image_t *pix;
  ^~~~~~~~~
/Users/jvavrek/STIR/src/display/display_array.cxx:234:3: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
  register  int y_count,x_count,i;
  ^~~~~~~~~
/Users/jvavrek/STIR/src/display/display_array.cxx:234:3: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
  register  int y_count,x_count,i;
  ^~~~~~~~~
/Users/jvavrek/STIR/src/display/display_array.cxx:234:3: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
  register  int y_count,x_count,i;
  ^~~~~~~~~
4 errors generated.
make[2]: *** [src/display/CMakeFiles/display.dir/display_array.cxx.o] Error 1
```

According to [this similar issue](https://github.com/danini/graph-cut-ransac/issues/23) it can apparently be removed without breaking anything.

--- a/src/display/display_array.cxx
+++ b/src/display/display_array.cxx
@@ -230,8 +230,8 @@ static void Array2DtoSCImg (
 		      int scale, double maxi)
 {
   image_t   *pimage;
-  register  image_t *pix;
-  register  int y_count,x_count,i;
+  image_t *pix;
+  int y_count,x_count,i;
 
   // length x,y of original image
   const int org_length_y = plane.get_length();

